spec/models/interactive_spec.rb

```diff
@@ -125,4 +125,19 @@
     it { expect(assignment.result).to be_blank }
   end
 
+  context 'on errored try with no query_result' do
+    before do
+      expect(bridge).to receive(:run_try!).and_return({status: :errored, result: ''})
+    end
+
+    before do
+      interactive.submit_try!(user, query: 'foo')
+    end
+
+    it { expect(assignment.queries).to eq ['foo'] }
+    it { expect(assignment.query_results).to eq [] }
```

Comment:
Can adding a query without a query result led to inconsistencies?

Reply:
https://github.com/mumuki/mumuki-laboratory/blob/06e42bb96b5d9ae429a872d3b0708b5c374590fc/app/models/assignment.rb#L30-L31

... yep.